### PR TITLE
[rllib] Add evaluation option to DQN agent

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

Long-standing feature request to allow evaluation with eps=0 in DQN/DDPG.

### CODE REVIEW

**Comment by ericl** [comment]
jenkins retest this please

File: python/ray/rllib/agents/dqn/dqn.py [hunk 1]
```diff
@@ -206,6 +229,12 @@ def _train(self):
                 "max_exploration": max(exp_vals),
                 "num_target_updates": self.num_target_updates,
             }, **self.optimizer.stats()))
+
+        if self.config["evaluation_interval"]:
+            if self.iteration % self.config["evaluation_interval"] == 0:
+                self.evaluation_metrics = self._evaluate()
+            result.update(self.evaluation_metrics)
```
Comment (richardliaw) [inline python/ray/rllib/agents/dqn/dqn.py]: Does this work in Tune? I think this might affect the CSV logger

Comment (richardliaw) [reply python/ray/rllib/agents/dqn/dqn.py]: (not sure but it's because the CSV logger is not robust to changing field sizes and probably should be fixed)

**Review by ericl — COMMENTED** [review]
It should be fine since the evaluation field is always present.

**Review by richardliaw — APPROVED** [review]
Looks good if this works with run_experiments

**Pull request merged**